zebra/zebra_rib.c

```diff
@@ -1142,8 +1147,10 @@ static struct route_entry *rib_choose_best(struct route_entry *current,
 	 * or loopback interface.  If not, pick the last connected
 	 * route of the set of lowest metric connected routes.
 	 */
-	if (alternate->type == ZEBRA_ROUTE_CONNECT) {
-		if (current->type != ZEBRA_ROUTE_CONNECT)
```

Comment:
This is treating `C` and `L` on the same level. I think we need to additionally have `L` above `C`?

Reply:
I need to abstract this chunk o change

Reply:
fixed